src/main/kotlin/org/rust/lang/core/stubs/StubImplementations.kt

```diff
@@ -1183,45 +1177,54 @@ class RsUnaryExprStub(
 
 }
 
-sealed class RsStubLiteralType(val typeOrdinal: Int) {
-    object Boolean : RsStubLiteralType(0)
-    class Char(val isByte: kotlin.Boolean) : RsStubLiteralType(1)
-    class String(val length: Long?, val isByte: kotlin.Boolean) : RsStubLiteralType(2)
-    class Integer(val kind: TyInteger?) : RsStubLiteralType(3)
-    class Float(val kind: TyFloat?) : RsStubLiteralType(4)
+sealed class RsStubLiteralKind(val kindOrdinal: Int) {
+    class Boolean(val value: kotlin.Boolean) : RsStubLiteralKind(0)
+    class Char(val value: kotlin.String?, val isByte: kotlin.Boolean) : RsStubLiteralKind(1)
+    class String(val value: kotlin.String?, val isByte: kotlin.Boolean) : RsStubLiteralKind(2)
+    class Integer(val value: Long?, val ty: TyInteger?) : RsStubLiteralKind(3)
+    class Float(val value: Double?, val ty: TyFloat?) : RsStubLiteralKind(4)
 
     companion object {
-        fun deserialize(dataStream: StubInputStream): RsStubLiteralType? {
-            with (dataStream) {
+        fun deserialize(dataStream: StubInputStream): RsStubLiteralKind? {
+            with(dataStream) {
                 val ordinal = readByte().toInt()
```

Comment:
`ordinal` can be moved inside `when`

Reply:
If move it to `when` it will be marked as unused because it is really not used in `when` branches. 
So we can even remove `ordinal` variable at all. But I'd prefer to keep this variable to simplify reading and debugging. So let's keep it as is to avoid unnecessary `unused variable` warning.
Btw IDEA doesn't suggest to move `ordinal` to `when`